sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/ReturnEmptyCollectionInsteadOfNull.cs

```diff
@@ -115,7 +117,9 @@ private static IEnumerable<ReturnStatementSyntax> GetReturnNullStatements(BlockS
         {
             return methodBlock.DescendantNodes()
                 .OfType<ReturnStatementSyntax>()
-                .Where(returnStatement => returnStatement.Expression.IsKind(SyntaxKind.NullLiteralExpression));
+                .Where(returnStatement =>
+                    returnStatement.Expression.IsKind(SyntaxKind.NullLiteralExpression) &&
+                    returnStatement.FirstAncestorOrSelf<ParenthesizedLambdaExpressionSyntax>() == null);
```

Comment:
I think fixes the immediate problem, but shouldn't we treat the lambdas as normal methods and forbid returning null from lambdas as well? Also probably the rule implementation will be simpler if we register for `return` statements and raise when they return null and their first parent of type Method, Property or ParenthesizedLambda is of type IEnumerable.

Reply:
You are right I will update the RSPEC and the rule.